gen_cdt_recipes.py

```diff
@@ -0,0 +1,189 @@
+import os
+import subprocess
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+import tqdm
+import click
+from ruamel.yaml import YAML
+
+from cdt_config import (
+    LEGACY_CDT_PATH,
+    LEGACY_CUSTOM_CDT_PATH,
+    CDT_PATH,
+    CUSTOM_CDT_PATH,
+)
+
+yaml = YAML()
+
+
+def _make_cdt_recipes(*, extra, cdt_path, arch_dist_tuples, cdts, exec):
+    futures = {}
+    for arch, dist in arch_dist_tuples:
+        for cdt, cfg in cdts.items():
+            if cfg["custom"]:
+                continue
+
+            _extra = extra + " --recursive"
+
+            futures[exec.submit(
+                subprocess.run,
+                (
+                    f"python rpm.py {cdt} --output-dir={cdt_path} "
+                    + f"--architecture={arch} --distro={dist} "
+                    + _extra
+                ),
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+                text=True,
+                shell=True
+            )] = {"cdt": cdt, "arch": arch, "dist": dist}
+    return futures
+
+
+def _cleanup_custom_cdt_overlaps(*, cdt_path, arch_dist_tuples, cdts):
+    for arch, dist in arch_dist_tuples:
+        for cdt, cfg in cdts.items():
+            if not cfg["custom"]:
+                continue
+
+            pth = os.path.join(
+                cdt_path,
+                cdt + "-" + dist.replace("ent", "") + "-" + arch,
+            )
+            if os.path.exists(pth):
+                try:
+                    subprocess.run(
+                        "git rm -r -f --ignore-unmatch " + pth,
```

Comment:
Would it make sense to use `gitpython` for this sort of thing?

Reply:
Sure. I don't want to spend time on that right now, but we can make that change later.

Reply:
issue here: https://github.com/conda-forge/cdt-builds/issues/4